task6/UPposts.js

```diff
@@ -0,0 +1,258 @@
+            photoLink: 'https://www.pressball.by/images/stories/2020/03/20200310231542.jpg'
+        },
+
+        {
+            id: '17',
+            description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+            createdAt: new Date('2020-03-25T23:00:00'),
+            author: 'Петр',
+            photoLink: 'https://www.pressball.by/images/stories/2020/03/20200310231542.jpg'
+        },
+
+        {
+            id: '18',
+            description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+            createdAt: new Date('2020-03-26T23:00:00'),
+            author: 'Петр',
+            photoLink: 'https://www.pressball.by/images/stories/2020/03/20200310231542.jpg'
+        },
+
+        {
+            id: '19',
+            description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+            createdAt: new Date('2020-03-27T23:00:00'),
+            author: 'Иванов Иван',
+            photoLink: 'https://www.pressball.by/images/stories/2020/03/20200310231542.jpg'
+        },
+
+        {
+            id: '20',
+            description: '[ss]',
+            createdAt: new Date('2020-03-28T23:00:00'),
+            author: 'Петр',
+            photoLink: 'https://www.pressball.by/images/stories/2020/03/20200310231542.jpg'
+        },
+
+    ];
+
+
+    function getPost(id) {
+        for (var i = 0; i < countPosts; i++) {
+            if (posts[i].id === id) {
+                return i;
+            }
+        }
+    }
+
+    function validatePost(post) {
+        if (typeof (post.id) !== 'string' || post.id === '' || typeof (post.description) !== 'string' || post.description.length > 200
+            || !(post.createdAt instanceof Date) || typeof (post.author) !== 'string' || post.author === '' || (typeof (post.photoLink) !== 'undefined' && typeof (post.photoLink) !== 'string')) {
+            return false;
+        }
+        return true;
+    }
+
+    function addPost(post) {
+        if (validatePost(post)) {
+            posts[countPosts] = post;
+            countPosts++;
+            return true;
+        }
+        return false;
+    }
+
+    function editPost(id, post) {
+        var ind = getPost(id);
+        var tmpPost = Object.assign({}, posts[ind]);
+        tmpPost.description = post.description;
+        tmpPost.photoLink = post.photoLink;
+        if (validatePost(tmpPost)) {
+            posts[ind] = tmpPost;
+            return true;
+        }
+        return false;
+
+    }
+
+    function removePost(id) {
+        var ind = getPost(id);
+        delete posts[ind];
+        countPosts--;
+    }
+
+    function getPosts(skip = 0, top = 10, filterConfig) {
```

Comment:
Filter method should be used for filtering
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort